examples/graphics/mesh-morph.html

```diff
@@ -0,0 +1,163 @@
+
+            // generate delta positions and normals - as morph targets store delta between base position / normal and modified position / normal
+            for (i = 0; i < modifiedNormals.length; i++) {
+                modifiedPositions[i] -= positions[i];
+                modifiedNormals[i] -= normals[i];
+            }
+
+            // create a morph target
+            return new pc.MorphTarget(app.graphicsDevice, {
+                deltaPositions: modifiedPositions,
+                deltaNormals: modifiedNormals
+            });
+        };
+
+        // create the base mesh - a sphere, with higher amount of vertices / triangles
+        var mesh = pc.createSphere(app.graphicsDevice, { latitudeBands: 200, longitudeBands: 200 });
+
+        // obtain base mesh vertex / index data
+        var srcPositions = [], srcNormals = [], indices = [];
+        mesh.getPositions(srcPositions);
+        mesh.getNormals(srcNormals);
+        mesh.getIndices(indices);
+
+        // build 3 targets by expanding a part of sphere along 3 planes, specified by the normal
+        var targets = [];
+        targets.push(createMorphTarget(srcPositions, srcNormals, indices, 1, 0, 0));
+        targets.push(createMorphTarget(srcPositions, srcNormals, indices, 0, 1, 0));
+        targets.push(createMorphTarget(srcPositions, srcNormals, indices, 0, 0, 1));
+
+        // create a morph using these 3 targets
+        mesh.morph = new pc.Morph(targets);
+
+        // Create the mesh instance
+        var node = new pc.GraphNode();
+        var material = new pc.StandardMaterial();
+        this.meshInstance = new pc.MeshInstance(node, mesh, material);
+
+        // Create a model and add the mesh instance to it
+        var model = new pc.Model();
+        model.graph = node;
+        model.meshInstances = [ this.meshInstance ];
+
+        // add morph instance - this is where currently set weights are stored
+        var morphInstance = new pc.MorphInstance(mesh.morph);
+        meshInstance.morphInstance = morphInstance;
+        model.morphInstances.push(morphInstance);
+
+        // Create Entity and add it to the scene
+        this.entity = new pc.Entity();
+        entity.setLocalPosition(0, 0, 0);
+        app.root.addChild(this.entity);
+
+        // Add a model compoonent
+        app.systems.model.addComponent(this.entity, {
+            type: 'asset'
+        });
+        this.entity.model.model = model;
+
+        // update function called once per frame
+        var self = this;
+        var time = 0;
+        app.on("update", function (dt) {
+            time += dt;
+
+            // modify weights of all 3 morph targets along some sin curve with different frequency
+            self.meshInstance.morphInstance.setWeight(0, Math.abs(Math.sin(time)));
+            self.meshInstance.morphInstance.setWeight(1, Math.abs(Math.sin(time * 0.3)));
+            self.meshInstance.morphInstance.setWeight(2, Math.abs(Math.sin(time * 0.7)));
```

Comment:
You can just do `morphInstance` instead of `self.meshInstance.morphInstance`.

Reply:
done, thanks!